backend/module/eCampApi/src/eCampApi/V1/Rpc/Register/RegisterController.php

```diff
@@ -15,29 +18,29 @@ public function __construct(RegisterService $registerService) {
         $this->registerService = $registerService;
     }
 
-    public function registerAction() {
+    public function registerAction(): ViewModel {
         /** @var Request $request */
         $request = $this->getRequest();
         $content = $request->getContent();
 
         $data = (null != $content) ? Json::decode($content) : [];
 
         if (empty($data->username)) {
-            return new ApiProblem(400, 'No username provided');
+            throw new InvalidArgumentException('No username provided');
         }
         if (empty($data->email)) {
-            return new ApiProblem(400, 'No eMail provided');
+            throw new InvalidArgumentException('No email provided');
         }
         if (empty($data->password)) {
-            return new ApiProblem(400, 'No password provided');
+            throw new InvalidArgumentException('No password provided');
         }
         $data->mailAddress = $data->email;
         $user = $this->registerService->register($data);
 
         if ($user instanceof ApiProblem) {
```

Comment:
we should throw an exception in the registerService, but we can do that later

Reply:
Good catch, thanks! Actually, this `if` can never be satisfied, now that `RegisterService#register` is declared to return `User`. I have removed it. The `RegisterService` does throw some exceptions under some circumstances already, such as `NoAccessException` when ACL fails, etc.